pkg/apply/prune/prune.go

```diff
@@ -248,6 +260,18 @@ func createPruneEvent(id object.ObjMetadata, obj *unstructured.Unstructured, op
 	}
 }
 
+// createDeleteEvent is a helper function to package a delete event.
+func createDeleteEvent(id object.ObjMetadata, obj *unstructured.Unstructured, op event.DeleteEventOperation) event.Event {
```

Comment:
How we do we create Delete events with errors?

Reply:
In `destroy`, we did not have specific delete event errors before; only generic error events. And the event channel prune/destroy transformer did not handle error events. So this functionality is the same as before.

Reply:
But I think we should fix this now. As far as I can tell, the code on lines [184](https://github.com/kubernetes-sigs/cli-utils/pull/366/files#diff-9d41d28d25555dfdd294c704d1577e1d7ecac3fd06fe94fb9e4df5a1c1409e2fR184) and [193](https://github.com/kubernetes-sigs/cli-utils/pull/366/files#diff-9d41d28d25555dfdd294c704d1577e1d7ecac3fd06fe94fb9e4df5a1c1409e2fR193) will lead to Prune events being created during destroy.

Reply:
Fixed. Also created `EventFactory` interface to clean up event generation.